src/bidiMapper/domains/script/Realm.ts

```diff
@@ -169,94 +141,32 @@ export class Realm {
       return deepSerializedValue as Script.RemoteValue;
     }
 
-    if (deepSerializedValue.type === 'node') {
-      if (Object.hasOwn(bidiValue, 'backendNodeId')) {
-        let navigableId = this.navigableId;
-        if (Object.hasOwn(bidiValue, 'loaderId')) {
-          // `loaderId` should be always there after ~2024-03-05, when
-          // https://crrev.com/c/5116240 reaches stable.
-          // TODO: remove the check after the date.
-          navigableId = bidiValue.loaderId;
-          delete bidiValue['loaderId'];
-        }
-        (deepSerializedValue as unknown as Script.SharedReference).sharedId =
-          SharedIdParser.getSharedId(
-            this.#getBrowsingContextId(navigableId),
-            navigableId,
-            bidiValue.backendNodeId,
-            this.#sharedIdWithFrame
-          );
-        delete bidiValue['backendNodeId'];
-      }
-      if (Object.hasOwn(bidiValue, 'children')) {
-        for (const i in bidiValue.children) {
-          bidiValue.children[i] = this.#deepSerializedToBiDi(
-            bidiValue.children[i],
-            internalIdMap
-          );
-        }
-      }
-      if (
-        Object.hasOwn(bidiValue, 'shadowRoot') &&
-        bidiValue.shadowRoot !== null
-      ) {
-        bidiValue.shadowRoot = this.#deepSerializedToBiDi(
-          bidiValue.shadowRoot,
-          internalIdMap
-        );
-      }
-      // `namespaceURI` can be is either `null` or non-empty string.
-      if (bidiValue.namespaceURI === '') {
-        bidiValue.namespaceURI = null;
-      }
-    }
-
     // Recursively update the nested values.
     if (
       ['array', 'set', 'htmlcollection', 'nodelist'].includes(
         deepSerializedValue.type
       )
     ) {
       for (const i in bidiValue) {
-        bidiValue[i] = this.#deepSerializedToBiDi(bidiValue[i], internalIdMap);
+        bidiValue[i] = this.serializeForBiDi(bidiValue[i], internalIdMap);
       }
     }
     if (['object', 'map'].includes(deepSerializedValue.type)) {
       for (const i in bidiValue) {
         bidiValue[i] = [
-          this.#deepSerializedToBiDi(bidiValue[i][0], internalIdMap),
-          this.#deepSerializedToBiDi(bidiValue[i][1], internalIdMap),
+          this.serializeForBiDi(bidiValue[i][0], internalIdMap),
+          this.serializeForBiDi(bidiValue[i][1], internalIdMap),
```

Comment:
I wonder in case of `window` realm, this would call window's `serializeForBiDi` or the base realm's